x-pack/libbeat/management/fleet/manager.go

```diff
@@ -122,38 +126,52 @@ func (cm *Manager) CheckRawConfig(cfg *common.Config) error {
 	return nil
 }
 
+// UpdateStatus updates the manager with the current status for the beat.
+func (cm *Manager) UpdateStatus(status management.Status, msg string) {
+	cm.lock.Lock()
+	if cm.status != status || cm.msg != msg {
+		cm.status = status
+		cm.msg = msg
+		cm.lock.Unlock()
+		cm.client.Status(statusToProtoStatus(status), msg)
```

Comment:
here we update status on the client which will result in some action either during checkin or in watcher after some timeout. 
if somebody reports failure i think we talked about re-starting application immediately, can you make sure this flow is valid?

Reply:
This flow is valid from the side of the Elastic Agent, as reporting Failed will cause the application to be killed and restarted. This is fully tested in covered in the Elastic Agent unit tests, both in the case that the application returns Failed over the protocol and if the application just crashes.